cdap-master/src/main/java/co/cask/cdap/data/runtime/main/ExploreCustomClassLoaderTwillRunnable.java

```diff
@@ -0,0 +1,189 @@
+  }
+
+  @Override
+  public void initialize(TwillContext context) {
+    super.initialize(context);
+    
+    if (getClass().getClassLoader() instanceof URLClassLoader) {
+      LOG.info("Using CustomResourcesClassLoader to load config files bundled with application.");
+      customClassLoader =
+        new CustomResourcesClassLoader(((URLClassLoader) getClass().getClassLoader()).getURLs(),
+                                       getClass().getClassLoader());
+    } else {
+      LOG.warn("Classloader is not URLCLassLoader, config files bundled with application might not be loaded.");
+      customClassLoader = getClass().getClassLoader();
+    }
+    
+    ClassLoader previousClassLoader = Thread.currentThread().getContextClassLoader();
+    Thread.currentThread().setContextClassLoader(customClassLoader);
+
+    try {
+      @SuppressWarnings("unchecked") 
+      Class<? extends TwillRunnable> twillRunnableClass = 
+        (Class<? extends TwillRunnable>) customClassLoader.loadClass(CLASS_NAME);
+      twillRunnable = newInstance(twillRunnableClass);
+      twillRunnable.initialize(context);
+    } catch (Exception e) {
+      throw Throwables.propagate(e);
+    } finally {
+      Thread.currentThread().setContextClassLoader(previousClassLoader);
+    }
+  }
+
+  @Override
+  public void handleCommand(Command command) throws Exception {
+    ClassLoader previousClassLoader = Thread.currentThread().getContextClassLoader();
+    Thread.currentThread().setContextClassLoader(customClassLoader);
+    try {
+      twillRunnable.handleCommand(command);
+    } finally {
+      Thread.currentThread().setContextClassLoader(previousClassLoader);
+    }
+  }
+
+  @Override
+  public void stop() {
+    ClassLoader previousClassLoader = Thread.currentThread().getContextClassLoader();
+    Thread.currentThread().setContextClassLoader(customClassLoader);
+    try {
+      twillRunnable.stop();
+    } finally {
+      Thread.currentThread().setContextClassLoader(previousClassLoader);
+    }
+  }
+
+  @Override
+  public void destroy() {
+    ClassLoader previousClassLoader = Thread.currentThread().getContextClassLoader();
+    Thread.currentThread().setContextClassLoader(customClassLoader);
+    try {
+      twillRunnable.destroy();
+    } finally {
+      Thread.currentThread().setContextClassLoader(previousClassLoader);
+    }
+  }
+
+  @Override
+  public void run() {
+    ClassLoader previousClassLoader = Thread.currentThread().getContextClassLoader();
+    Thread.currentThread().setContextClassLoader(customClassLoader);
+    try {
+      twillRunnable.run();
+    } finally {
+      Thread.currentThread().setContextClassLoader(previousClassLoader);
+    }
+  }
+
+  @VisibleForTesting
+  static TwillRunnable newInstance(Class<? extends TwillRunnable> twillRunnableClass)
+    throws NoSuchMethodException, IllegalAccessException, InvocationTargetException, InstantiationException {
+    Constructor<? extends TwillRunnable> constructor = 
```

Comment:
To make this class more generic (i.e. be able to do the same classloading trick for all possible TwillRunnable), you can use the InstantiatorFactory instead (which is exactly the same as what twill does).

```java
return new InstantiatorFactory(false).get(TypeToken.of(twillRunnableClass)).create();
```